### ISSUE TITLE
ppc64le: //tensorflow/contrib/lite/kernels:resize_bilinear_test test fails CPU test

### ISSUE THREAD

**Issue by wdirons** [tensorflow/tensorflow#20361]
Please assign this issue to me and add the tag: stat:community support

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: ppc64le Ubuntu 16.04.4
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: master from June 27th
- **Python version**: 2.7.12
- **Bazel version (if compiling from source)**: 0.11.0
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**:
bazel test -c opt --cache_test_results=no //tensorflow/contrib/lite/kernels:resize_bilinear_test

### Describe the problem
```
FAIL: //tensorflow/contrib/lite/kernels:resize_bilinear_test (see /root/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/ppc-opt/testlogs/tensorflow/contrib/lite/kernels/resize_bilinear_test/test.log)
INFO: From Testing //tensorflow/contrib/lite/kernels:resize_bilinear_test:
==================== Test output for //tensorflow/contrib/lite/kernels:resize_bilinear_test:
[==========] Running 10 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 10 tests from ResizeBilinearOpTest
[ RUN      ] ResizeBilinearOpTest.HorizontalResize
[       OK ] ResizeBilinearOpTest.HorizontalResize (0 ms)
[ RUN      ] ResizeBilinearOpTest.HorizontalResize8Bit
[       OK ] ResizeBilinearOpTest.HorizontalResize8Bit (0 ms)
[ RUN      ] ResizeBilinearOpTest.VerticalResize
[       OK ] ResizeBilinearOpTest.VerticalResize (0 ms)
[ RUN      ] ResizeBilinearOpTest.VerticalResize8Bit
[       OK ] ResizeBilinearOpTest.VerticalResize8Bit (0 ms)
[ RUN      ] ResizeBilinearOpTest.TwoDimensionalResize
[       OK ] ResizeBilinearOpTest.TwoDimensionalResize (0 ms)
[ RUN      ] ResizeBilinearOpTest.TwoDimensionalResize8Bit
[       OK ] ResizeBilinearOpTest.TwoDimensionalResize8Bit (1 ms)
[ RUN      ] ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches
[       OK ] ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches (0 ms)
[ RUN      ] ResizeBilinearOpTest.ThreeDimensionalResize
[       OK ] ResizeBilinearOpTest.ThreeDimensionalResize (0 ms)
[ RUN      ] ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches8Bit
tensorflow/contrib/lite/kernels/resize_bilinear_test.cc:261: Failure
Value of: m.GetOutput<uint8>()
Expected: has 18 elements where
element #0 is approximately 3 (absolute error <= 9.9999997e-06),
element #1 is approximately 5 (absolute error <= 9.9999997e-06),
element #2 is approximately 6 (absolute error <= 9.9999997e-06),
element #3 is approximately 7 (absolute error <= 9.9999997e-06),
element #4 is approximately 9 (absolute error <= 9.9999997e-06),
element #5 is approximately 10 (absolute error <= 9.9999997e-06),
element #6 is approximately 9 (absolute error <= 9.9999997e-06),
element #7 is approximately 11 (absolute error <= 9.9999997e-06),
element #8 is approximately 12 (absolute error <= 9.9999997e-06),
element #9 is approximately 4 (absolute error <= 9.9999997e-06),
element #10 is approximately 8 (absolute error <= 9.9999997e-06),
element #11 is approximately 10 (absolute error <= 9.9999997e-06),
element #12 is approximately 8 (absolute error <= 9.9999997e-06),
element #13 is approximately 12 (absolute error <= 9.9999997e-06),
element #14 is approximately 14 (absolute error <= 9.9999997e-06),
element #15 is approximately 10 (absolute error <= 9.9999997e-06),
element #16 is approximately 13 (absolute error <= 9.9999997e-06),
element #17 is approximately 16 (absolute error <= 9.9999997e-06)
  Actual: { '\x3' (3), '\x5' (5), '\x6' (6), '\a' (7), '\t' (9), '\n' (10, 0xA), '\t' (9), '\v' (11, 0xB), '\f' (12, 0xC), '\x4' (4), '\b' (8), '\n' (10, 0xA), '\b' (8), '\f' (12, 0xC), '\xE' (14), '\n' (10, 0xA), '\xE' (14), '\x10' (16) }, whose element #16 doesn't match, which is 1 from 13
tensorflow/contrib/lite/kernels/resize_bilinear_test.cc:278: Failure
Value of: const_m.GetOutput<uint8>()
Expected: has 18 elements where
element #0 is approximately 3 (absolute error <= 9.9999997e-06),
element #1 is approximately 5 (absolute error <= 9.9999997e-06),
element #2 is approximately 6 (absolute error <= 9.9999997e-06),
element #3 is approximately 7 (absolute error <= 9.9999997e-06),
element #4 is approximately 9 (absolute error <= 9.9999997e-06),
element #5 is approximately 10 (absolute error <= 9.9999997e-06),
element #6 is approximately 9 (absolute error <= 9.9999997e-06),
element #7 is approximately 11 (absolute error <= 9.9999997e-06),
element #8 is approximately 12 (absolute error <= 9.9999997e-06),
element #9 is approximately 4 (absolute error <= 9.9999997e-06),
element #10 is approximately 8 (absolute error <= 9.9999997e-06),
element #11 is approximately 10 (absolute error <= 9.9999997e-06),
element #12 is approximately 8 (absolute error <= 9.9999997e-06),
element #13 is approximately 12 (absolute error <= 9.9999997e-06),
element #14 is approximately 14 (absolute error <= 9.9999997e-06),
element #15 is approximately 10 (absolute error <= 9.9999997e-06),
element #16 is approximately 13 (absolute error <= 9.9999997e-06),
element #17 is approximately 16 (absolute error <= 9.9999997e-06)
  Actual: { '\x3' (3), '\x5' (5), '\x6' (6), '\a' (7), '\t' (9), '\n' (10, 0xA), '\t' (9), '\v' (11, 0xB), '\f' (12, 0xC), '\x4' (4), '\b' (8), '\n' (10, 0xA), '\b' (8), '\f' (12, 0xC), '\xE' (14), '\n' (10, 0xA), '\xE' (14), '          \x10' (16) }, whose element #16 doesn't match, which is 1 from 13
[  FAILED  ] ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches8Bit (0 ms)
[ RUN      ] ResizeBilinearOpTest.ThreeDimensionalResize8Bit
tensorflow/contrib/lite/kernels/resize_bilinear_test.cc:293: Failure
Value of: m.GetOutput<uint8>()
Expected: has 18 elements where
element #0 is approximately 3 (absolute error <= 9.9999997e-06),
element #1 is approximately 4 (absolute error <= 9.9999997e-06),
element #2 is approximately 5 (absolute error <= 9.9999997e-06),
element #3 is approximately 8 (absolute error <= 9.9999997e-06),
element #4 is approximately 6 (absolute error <= 9.9999997e-06),
element #5 is approximately 10 (absolute error <= 9.9999997e-06),
element #6 is approximately 7 (absolute error <= 9.9999997e-06),
element #7 is approximately 8 (absolute error <= 9.9999997e-06),
element #8 is approximately 9 (absolute error <= 9.9999997e-06),
element #9 is approximately 12 (absolute error <= 9.9999997e-06),
element #10 is approximately 10 (absolute error <= 9.9999997e-06),
element #11 is approximately 14 (absolute error <= 9.9999997e-06),
element #12 is approximately 9 (absolute error <= 9.9999997e-06),
element #13 is approximately 10 (absolute error <= 9.9999997e-06),
element #14 is approximately 11 (absolute error <= 9.9999997e-06),
element #15 is approximately 13 (absolute error <= 9.9999997e-06),
element #16 is approximately 12 (absolute error <= 9.9999997e-06),
element #17 is approximately 16 (absolute error <= 9.9999997e-06)
  Actual: { '\x3' (3), '\x4' (4), '\x5' (5), '\b' (8), '\x6' (6), '\n' (10, 0xA), '\a' (7), '\b' (8), '\t' (9), '\f' (12, 0xC), '\n' (10, 0xA), '\xE' (14), '\t' (9), '\n' (10, 0xA), '\v' (11, 0xB), '\xE' (14), '\f' (12, 0xC), '          \x10' (16) }, whose element #15 doesn't match, which is 1 from 13
tensorflow/contrib/lite/kernels/resize_bilinear_test.cc:305: Failure
Value of: const_m.GetOutput<uint8>()
Expected: has 18 elements where
element #0 is approximately 3 (absolute error <= 9.9999997e-06),
element #1 is approximately 4 (absolute error <= 9.9999997e-06),
element #2 is approximately 5 (absolute error <= 9.9999997e-06),
element #3 is approximately 8 (absolute error <= 9.9999997e-06),
element #4 is approximately 6 (absolute error <= 9.9999997e-06),
element #5 is approximately 10 (absolute error <= 9.9999997e-06),
element #6 is approximately 7 (absolute error <= 9.9999997e-06),
element #7 is approximately 8 (absolute error <= 9.9999997e-06),
element #8 is approximately 9 (absolute error <= 9.9999997e-06),
element #9 is approximately 12 (absolute error <= 9.9999997e-06),
element #10 is approximately 10 (absolute error <= 9.9999997e-06),
element #11 is approximately 14 (absolute error <= 9.9999997e-06),
element #12 is approximately 9 (absolute error <= 9.9999997e-06),
element #13 is approximately 10 (absolute error <= 9.9999997e-06),
element #14 is approximately 11 (absolute error <= 9.9999997e-06),
element #15 is approximately 13 (absolute error <= 9.9999997e-06),
element #16 is approximately 12 (absolute error <= 9.9999997e-06),
element #17 is approximately 16 (absolute error <= 9.9999997e-06)
  Actual: { '\x3' (3), '\x4' (4), '\x5' (5), '\b' (8), '\x6' (6), '\n' (10, 0xA), '\a' (7), '\b' (8), '\t' (9), '\f' (12, 0xC), '\n' (10, 0xA), '\xE' (14), '\t' (9), '\n' (10, 0xA), '\v' (11, 0xB), '\xE' (14), '\f' (12, 0xC), '          \x10' (16) }, whose element #15 doesn't match, which is 1 from 13
[  FAILED  ] ResizeBilinearOpTest.ThreeDimensionalResize8Bit (1 ms)
[----------] 10 tests from ResizeBilinearOpTest (2 ms total)

[----------] Global test environment tear-down
[==========] 10 tests from 1 test case ran. (2 ms total)
[  PASSED  ] 8 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches8Bit
[  FAILED  ] ResizeBilinearOpTest.ThreeDimensionalResize8Bit

 2 FAILED TESTS
================================================================================
```

### Source code / logs
see above

**Comment by sandipmgiri** [comment]
@wdirons , this test is not failing for me on ppc64le (CPU setup). I'm using community build mechanism and running the tests :

```
$  bazel test -c opt //tensorflow/contrib/lite/kernels:resize_bilinear_test 
INFO: Analysed target //tensorflow/contrib/lite/kernels:resize_bilinear_test (11 packages loaded).
INFO: Found 1 test target...
Target //tensorflow/contrib/lite/kernels:resize_bilinear_test up-to-date:
  bazel-bin/tensorflow/contrib/lite/kernels/resize_bilinear_test
INFO: Elapsed time: 5.103s, Critical Path: 0.79s
INFO: Build completed successfully, 1 total action
//tensorflow/contrib/lite/kernels:resize_bilinear_test          (cached) PASSED in 0.8s
```

**Comment by npanpaliya** [comment]
@sandipmgiri - The test passes if you use 1.9.0rc0 tag because it didn't have support for uint8_t datatype, moreover the test file resize_bilinear_test.cc itself didn't include any tests for uint8_t datatype. The support for generic datatype was added later in the commit https://github.com/tensorflow/tensorflow/commit/4f912021b04f5f82b0d1a6bba5b32a24d7cb9fca and so are the new tests in the resize_bilinear_test which are failing for me on ppc64le.

**Comment by npanpaliya** [comment]
Found the cause of the issue. Working to fix the problem.
The way float value is type casted to uint8_t is the root cause of the problem on ppc64le at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h#L5506.

**Comment by npanpaliya** [comment]
Further debugging this on both x86 and ppc64le, I found some weirdness on x86 and ppc64le behaving appropriately -
Below is the code snippet from optimized_ops.h which is executed when testcase "ResizeBilinearOpTest.TwoDimensionalResizeWithTwoBatches8Bit" is run.
`template <typename T>
inline void ResizeBilinearGenericSmallChannel(
    const T* input_data, const Dims<4>& input_dims, T* output_data,
    const Dims<4>& output_dims, int32 batches, int32 input_height,
    int32 input_width, int32 depth, int32 output_height, int32 output_width,
    float height_scale, float width_scale) {
   ...
          for (int32 d = 0; d < depth; d++) {
          const T* input_ptr = &input_data[d];
          float fOutput = input_ptr[input_offset[0]] * scale[0] +
                                         input_ptr[input_offset[1]] * scale[1] +
                                         input_ptr[input_offset[2]] * scale[2] +
                                         input_ptr[input_offset[3]] * scale[3];
          *output_ptr = static_cast<T>(fOutput);
          LOG(INFO) << "Uint8 Output:  " << +*output_ptr;
          LOG(INFO) << "Float Output:  " << fOutput;
          output_ptr++;          
        }
   ...
  }
` 
For the mentioned  testcase, the above for loop runs for 18 times. For the 17th item, I get different values for uint8 and float on x86. Below are the logs I printed for the mentioned test case -

On ppc64le -
`2018-07-03 12:32:46.008931: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  3
2018-07-03 12:32:46.008937: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  3
2018-07-03 12:32:46.008942: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  5
2018-07-03 12:32:46.008948: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  5
2018-07-03 12:32:46.008953: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  6
2018-07-03 12:32:46.008959: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  6
2018-07-03 12:32:46.008964: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  7
2018-07-03 12:32:46.008969: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  7
2018-07-03 12:32:46.008974: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  9
2018-07-03 12:32:46.008980: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  9
2018-07-03 12:32:46.008985: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  10
2018-07-03 12:32:46.008992: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  10
2018-07-03 12:32:46.008997: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  9
2018-07-03 12:32:46.009003: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  9
2018-07-03 12:32:46.009008: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  11
2018-07-03 12:32:46.009013: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  11
2018-07-03 12:32:46.009019: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  12
2018-07-03 12:32:46.009024: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  12
2018-07-03 12:32:46.009029: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  4
2018-07-03 12:32:46.009035: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  4
2018-07-03 12:32:46.009040: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  8
2018-07-03 12:32:46.009046: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  8
2018-07-03 12:32:46.009051: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  10
2018-07-03 12:32:46.009057: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  10
2018-07-03 12:32:46.009062: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  8
2018-07-03 12:32:46.009067: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  8
2018-07-03 12:32:46.009072: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  12
2018-07-03 12:32:46.009078: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  12
2018-07-03 12:32:46.009083: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  14
2018-07-03 12:32:46.009089: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  14
2018-07-03 12:32:46.009096: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  10
2018-07-03 12:32:46.009101: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  10
2018-07-03 12:32:46.009106: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  14
2018-07-03 12:32:46.009112: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  14
2018-07-03 12:32:46.009117: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5522] Uint8 Output:  16
2018-07-03 12:32:46.009123: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5524] Float Output:  16
`

On x86 -
`2018-07-03 12:32:33.869341: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  3
2018-07-03 12:32:33.869435: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  3
2018-07-03 12:32:33.869449: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  5
2018-07-03 12:32:33.869458: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  5
2018-07-03 12:32:33.869468: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  6
2018-07-03 12:32:33.869476: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  6
2018-07-03 12:32:33.869484: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  7
2018-07-03 12:32:33.869492: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  7
2018-07-03 12:32:33.869501: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  9
2018-07-03 12:32:33.869509: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  9
2018-07-03 12:32:33.869518: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  10
2018-07-03 12:32:33.869527: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  10
2018-07-03 12:32:33.869536: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  9
2018-07-03 12:32:33.869544: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  9
2018-07-03 12:32:33.869553: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  11
2018-07-03 12:32:33.869561: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  11
2018-07-03 12:32:33.869569: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  12
2018-07-03 12:32:33.869578: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  12
2018-07-03 12:32:33.869587: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  4
2018-07-03 12:32:33.869595: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  4
2018-07-03 12:32:33.869603: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  8
2018-07-03 12:32:33.869611: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  8
2018-07-03 12:32:33.869620: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  10
2018-07-03 12:32:33.869628: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  10
2018-07-03 12:32:33.869637: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  8
2018-07-03 12:32:33.869645: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  8
2018-07-03 12:32:33.869653: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  12
2018-07-03 12:32:33.869661: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  12
2018-07-03 12:32:33.869670: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  14
2018-07-03 12:32:33.869678: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  14
2018-07-03 12:32:33.869687: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  10
2018-07-03 12:32:33.869695: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  10
2018-07-03 12:32:33.869713: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  13
2018-07-03 12:32:33.869723: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  14
2018-07-03 12:32:33.869732: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5519] Uint8 Output:  16
2018-07-03 12:32:33.869740: I ./tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h:5521] Float Output:  16
`

And the test case's expected output is designed as per uint8 value which is 13 at the index 17. However, the same test case with same inputs is also written for float datatype and it has 14 at the 17th index in the expected output. 
Looks like either it is some casting issue on x86 or I'm missing something here. Could anyone please help me?

**Comment by wdirons** [comment]
@npanpaliya , I'm not sure how the math worked in this case, but it does seem odd that the expected results are different if using float vs uint8. Maybe this is due to a rounding error. Perhaps you can attempt to change the input of the testcase to values where the results for float and uint8 are the same?

**Comment by npanpaliya** [comment]
@wdirons - I tried with different input values and fortunately for those values, I see x86 and ppc64le are giving same results. But not sure if Community can accept it as a fix.
@rohan100jain and others from TF Community - Request you to provide the inputs on this. Will it be okay if we just change the input values to the testcase, such that the output is same on both the platforms?

**Comment by wdirons** [comment]
I think the question is more why for x86 are we expecting different results when casting to uint8 then when using float.

**Comment by npanpaliya** [comment]
@wdirons - Raised a PR for this by changing input/output for the failing tests - https://github.com/tensorflow/tensorflow/pull/20744

**Comment by wdirons** [comment]
Fixed with PR #20744
